examples/step-77/step-77.cc

```diff
@@ -554,77 +557,130 @@ namespace Step77
+            // All four of these operations are represented by member variables
+            // of the SUNDIALS::KINSOL class that are of type `std::function`,
+            // i.e., they are objects to which we can assign a pointer to a
+            // function or, as we do here, a "lambda function" that takes the
+            // appropriate arguments and returns the appropriate information. By
+            // convention, KINSOL wants that functions doing something
+            // nontrivial return an integer where zero indicates success. It
+            // turns out that we can do all of this in just 25 lines of code.
+            //
+            // (If you're not familiar what "lambda functions" are, take
+            // a look at step-12 or at the
+            // [wikipedia
+            // page](https://en.wikipedia.org/wiki/Anonymous_function) on the
+            // subject. The idea of lambda functions is that one wants to define
+            // a function with a certain set of arguments, but (i) not make it a
+            // named functions because, typically, the function is used in only
+            // one place and it seems unnecessary to give it a global name; and
+            // (ii) that the function has access to some of the variables that
+            // exist at the place where it is defined, including member
+            // variables. The syntax of lambda functions is awkward, but
+            // ultimately quite useful.)
+            //
+            // At the very end of the code block we then tell KINSOL to go to
+            // work and solve our problem. The member functions called from the
+            // 'residual', 'setup_jacobian', and 'solve_jacobian_system'
+            // functions will then print output to screen that allows us to
+            // follow along with the progress of the program.
+            nonlinear_solver.reinit_vector = [&](Vector<double> &x) {
+              x.reinit(dof_handler.n_dofs());
+            };
+
+            nonlinear_solver.residual =
+              [&](const Vector<double> &evaluation_point,
+                  Vector<double> &      residual) {
+                compute_residual(evaluation_point, residual);
+
+                return 0;
+              };
+
+            nonlinear_solver.setup_jacobian =
+              [&](const Vector<double> &current_u,
+                  const Vector<double> & /*current_f*/) {
+                compute_and_factorize_jacobian(current_u);
+
+                return 0;
+              };
+
+            nonlinear_solver.solve_with_jacobian =
+              [&](const Vector<double> &rhs,
+                  Vector<double> &      dst,
+                  const double          tolerance) {
+                this->solve(rhs, dst, tolerance);
+
+                return 0;
+              };
+
+            nonlinear_solver.solve(current_solution);
+          }
+        else
+#endif
+#ifdef DEAL_II_WITH_TRILINOS
+          if (non_liner_solver_name == "NOX")
```

Comment:
@bangerth I would like to make NOX an option in step-77. Do you agree?

Reply:
That would be a very cool idea.

Reply:
Yes, I would love that!